### DISCUSSION TITLE
Zenith module repos: (1 repo has many modules) or (1 repo has 1 module)?

### DISCUSSION THREAD

**Discussion by maert** [dagger/dagger#5818] [Ideas]
### There are experiments happening around the repos structure for Zenith modules.

A couple of popular patterns are:

https://github.com/shykes/daggerverse
https://github.com/kpenfound/dagger-modules
with many modules under one repo

and

https://github.com/jpadams/dagger-module-trivy
with one module per repo

**Comment by shykes** [comment]
I am definitely a big proponent of the "daggerverse repo". I like that we're experimenting with different models; and I like that we can accomodate them all; but definitely think the daggerverse repo model is superior, and will gain popularity.

**Comment by nipuna-perera** [comment]
I agree with Solomon. In fact that's a model we use today with our Jenkins
shared libraries. All the functions are in the same repo. I think that's
will suit companies who want to maintain a single repo for all their
internal dagger modules.

Honestly with that, both models are supported as you could technically have
a single module per repo if you want.
